instrumentation/github.com/aws/aws-lambda-go/otellambda/config.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otellambda
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// A Flusher dictates how the instrumentation will attempt to flush
+// unexported spans at the end of each Lambda innovation. This is
+// very important in asynchronous settings because the Lambda runtime
+// may enter a 'frozen' state any time after the invocation completes.
+// Should this freeze happen and spans are left unexported, there can be a
+// long delay before those spans are exported.
+type Flusher interface {
+	ForceFlush(context.Context) error
+}
+
+type noopFlusher struct{}
+
+func (*noopFlusher) ForceFlush(context.Context) error { return nil }
+
+// Compile time check our noopFlusher implements Flusher
+var _ Flusher = &noopFlusher{}
+
+// An EventToTextMapCarrierConverter defines how the instrumentation should
+// prepare a TextMapCarrier for the configured propagator to read from. This
+// extra step is necessary because Lambda does not have HTTP headers to read
+// from and instead stores the headers it was invoked with (including TraceID, etc.)
+// as part of the invocation event. If using the AWS XRay tracing then the
+// trace information is instead stored in the Lambda environment.
+type EventToTextMapCarrierConverter func(eventJSON []byte) propagation.TextMapCarrier
```

Comment:
This name is a mouthful!  Is there something shorter that would work just as well?

Reply:
Going to leave some brainstorming ideas here and come back to them later:

- TraceDetector: **Pros**: detects whether there is a trace in the event/environ, short and intuitive name **Cons**: don't want to create confusion with ResourceDetectors, sounds like a struct not function
- EventToCarrier: **Pros** describes datatypes in/out, short **Cons** desired outcome is not intuitive
- GetTraceHeader: **Pros** describes purpose of function, intuitive output **Cons** Implies trace is in a Header obj somewhere (not always true eg xray)

Reply:
`EventToCarrier` LGTM